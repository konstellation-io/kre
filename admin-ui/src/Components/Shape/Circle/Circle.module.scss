@import 'Styles/colors';
@import 'Styles/mixins';

.loaderContainer {
  position: relative;
  margin: auto;
}

.innerContainer {
  @include font-small;
  color: font-color(light);

  background-color: transparent;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.initializing {
  opacity: 0;
  animation-name: showText;
  animation-duration: 1s;
  animation-delay: 1s;
  animation-fill-mode: forwards;
}

@keyframes showText {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
