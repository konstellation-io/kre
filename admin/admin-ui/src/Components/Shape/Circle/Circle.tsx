import React, { FC } from 'react';

import Lottie from '../../Lottie/Lottie';
import { STATES } from 'Constants/application';
import animationData from './Circle.json';
import cx from 'classnames';
import styles from './Circle.module.scss';

const ANIM_SEGMENTS: { [key: string]: [number, number] } = {
  [STATES.INITIALIZING]: [0, 119],
  [STATES.LOADING]: [120, 399],
  [STATES.DISABLED]: [400, 799],
  [STATES.ERROR]: [800, 1199],
  [STATES.ALERT]: [1200, 1599],
  [STATES.SUCCESS]: [1600, 1999]
};

type Props = {
  animation?: string;
  label?: string;
  children?: JSX.Element;
  size?: number;
};

const Circle: FC<Props> = ({
  animation = STATES.INITIALIZING,
  label = 'LOADING...',
  size = 250,
  children
}) => {
  const animationSegments = [ANIM_SEGMENTS[animation]];

  // Adds loading animation after finishing initiallization animation.
  if (animation === STATES.INITIALIZING) {
    animationSegments.push(ANIM_SEGMENTS.LOADING);
  }

  return (
    <div
      className={styles.loaderContainer}
      style={{ width: size, height: size }}
    >
      <Lottie
        options={{ animationData }}
        width={size}
        height={size}
        segments={animationSegments}
        forceSegments
      />
      <div
        className={cx(styles.innerContainer, {
          [styles.initializing]: animation === STATES.INITIALIZING
        })}
      >
        {children || label}
      </div>
    </div>
  );
};

export default Circle;
