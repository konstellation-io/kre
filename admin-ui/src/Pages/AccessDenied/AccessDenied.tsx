import Circle from 'Components/Shape/Circle/Circle';
import React from 'react';
import { STATES } from 'Constants/application';
import styles from './AccessDenied.module.scss';

function AccessDenied() {
  return (
    <div className={styles.bg}>
      <div className={styles.grid}>
        <div className={styles.container}>
          <h1>Higher access level required</h1>
          <Circle animation={STATES.ERROR} label="ACCESS DENIED" />
        </div>
      </div>
    </div>
  );
}

export default AccessDenied;
