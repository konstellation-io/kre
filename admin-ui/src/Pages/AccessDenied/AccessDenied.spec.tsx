import AccessDenied from './AccessDenied';
import Circle from 'Components/Shape/Circle/Circle';
import React from 'react';
import { shallow } from 'enzyme';

describe('AccessDenied', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AccessDenied />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.find('h1').text()).toBe('Higher access level required');
    expect(wrapper.exists(Circle)).toBeTruthy();
  });
});
