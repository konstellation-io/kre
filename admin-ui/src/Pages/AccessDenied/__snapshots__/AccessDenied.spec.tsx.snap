// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessDenied matches snapshot 1`] = `
<div
  className="bg"
>
  <div
    className="grid"
  >
    <div
      className="container"
    >
      <h1>
        Higher access level required
      </h1>
      <Circle
        animation="ERROR"
        label="ACCESS DENIED"
      />
    </div>
  </div>
</div>
`;
